src/swap/handle_check_address.c

```diff
@@ -13,88 +13,94 @@
+                                         size_t hash_len) {
+    uint8_t bip32_path_len;
+    uint32_t bip32_path[MAX_BIP32_PATH];
+    bool ret = false;
+    buffer_t cdata = {
+        .ptr = params->address_parameters,
+        .size = params->address_parameters_length,
+        .offset = 0UL,
+    };
+    pubkey_ctx_t pk_info = {
+        .subwallet_id = 698983191,
+        .is_v3r2 = false,
+    };
 
     if (params->address_parameters == NULL) {
         PRINTF("derivation path expected\n");
-        return;
+        return false;
     }
+
     PRINTF("address_parameters %.*H\n",
            params->address_parameters_length,
            params->address_parameters);
 
-    if (params->address_to_check == NULL) {
-        PRINTF("Address to check expected\n");
-        return;
-    }
-    PRINTF("Address to check %s\n", params->address_to_check);
-    if (strlen(params->address_to_check) != ADDRESS_BASE64_LENGTH) {
-        PRINTF("Address to check expected length %d, not %d\n",
-               ADDRESS_BASE64_LENGTH,
-               strlen(params->address_to_check));
-        return;
-    }
-
-    buffer_t cdata;
-    cdata.ptr = params->address_parameters;
-    cdata.size = params->address_parameters_length;
-    cdata.offset = 0;
-
-    uint8_t bip32_path_len;
-    uint32_t bip32_path[MAX_BIP32_PATH];
-    pubkey_ctx_t pk_info;
-    pk_info.subwallet_id = 698983191;
-    pk_info.is_v3r2 = false;
     if (get_public_key_helper(0, &cdata, &bip32_path_len, bip32_path, &pk_info) != 0) {
         PRINTF("Failed to read public key\n");
-        return;
+        return false;
     }
+
     PRINTF("Derived on path %.*H\n", 4 * bip32_path_len, bip32_path);
     PRINTF("Public key %.*H\n", PUBKEY_LEN, pk_info.raw_public_key);
 
-    uint8_t hash[HASH_LEN] = {0};
-    if (!pubkey_to_hash(pk_info.raw_public_key,
-                        pk_info.subwallet_id,
-                        pk_info.is_v3r2,
-                        hash,
-                        sizeof(hash))) {
-        return;
-    }
-    PRINTF("hash %.*H\n", HASH_LEN, hash);
+    ret = pubkey_to_hash(pk_info.raw_public_key,
+                         pk_info.subwallet_id,
+                         pk_info.is_v3r2,
+                         hash,
+                         hash_len);
+    return ret;
+}
 
-    char address_base64url[ADDRESS_BASE64_LENGTH];
-    if (base64_to_base64url(params->address_to_check,
-                            ADDRESS_BASE64_LENGTH,
-                            address_base64url,
-                            ADDRESS_BASE64_LENGTH) < 0) {
+/**
+ * @brief Decode the given address in base64 to binary
+ *
+ * Internally convert base64 address to base64 url and then decode base64 url.APPNAME
+ *
+ * @param address BASE64 address to decode
+ * @param decoded decode buffer
+ * @param size decode buffer size
+ *
+ * @return true if success, false otherwise
+ */
+bool swap_decode_address(const char *address, uint8_t *decoded, size_t size) {
```

Comment:
You can use hardcoded array value if you are certain of the size (it's the case)
`bool swap_decode_address(const char *address, uint8_t decoded[ADDRESS_DECODED_LENGTH}) {`
It's given as ref also

Reply:
Done